Adjust Identity X events emitter

B77Mills opened the pull request:
Adjust how Identity X event emitter handles loginSource, source & actionSource.  
```js
const source = this.loginSource || this.source;
const dataActionSource = data ? data.actionSource : undefined;
const actionSource = dataActionSource || window.IdentityX.getLoginSource() || source;
```
It will then also set actionSource, loginSource & source to the calculated actionSource. 

I also add new mixin to the identityX package to handle the event for the auto signup from additionalEvent Data.  This specifically deals with the identity-x-auto-signup event.

B77Mills committed:
Ensure actionSource is handled in a consistent way

Ensure that actionSource, and related values(loginSource, source), are consistently being set/handled when emitting identity-x events.

B77Mills committed:
Set login source off response from autheticate

Set the loginSource from the res.data on authenticate to ensure when the authenticated event is emitted the actionSource prop is accounted for correctly.

B77Mills committed:
Stop sending identity-x events to windwo.dataLayer

B77Mills committed:
Add authentication actionSource support

onAuthenticate ensure actionSource is set off of the res.data before emitting the authenticated event and add check to set actionSource off data if set.  This will ensure it is set correctly for authentication instead of ensuring the __idx_ctx is set prior to emitting the authenticated event.

B77Mills committed:
Revert "Stop sending identity-x events to windwo.dataLayer"

This reverts commit eb87a82f403ca072de00c0d191fe739f5998d685.

B77Mills committed:
Add auto-signup event emitter/handler support

On login, authenticate & profile update add new global auto signup event emitter to trigger events for identity-x-auto-signup, normally with a { userID, productId, loginSource} as its payload.

solocommand reviewed (changes requested):
Don't create a getAsArray util here, and no need to import `object-path`. We're already importing `@parameter1/base-cms-object-path`, just use that.

The IdentityX event registrations should move from the `marko-web-gtm` package into the `marko-web-identity-x` package, and the override of `loginSource` to the dataLayer should be removed.

Do we need to distinguish between an auto-signup emitted from login vs authenticate? If so, we should either pass the event name as part of the payload, or send distinct events (`identity-x-auto-signup-from-login` vs `identity-x-auto-signup-from-authenticate`).

Looks good otherwise!

B77Mills committed:
Use @parameter1/base-cms-object-pat getAs Array

B77Mills committed:
Move DataLayer EventBus hook from gtm to identityX package

Move the call to add the EventBuss.$on event listener to push stuff to the data layer out of the gtm package and into the identityX pacakge

solocommand commented on packages/marko-web-identity-x/browser/mixins/global-auto-signup-event-emitter.js:
```diff
@@ -0,0 +1,18 @@
+import getAsArray from '@parameter1/base-cms-object-path';
```
Pretty sure this should be `import { getAsArray } ...` :)

B77Mills committed:
Just spred args on to the event

B77Mills committed:
Revert "Use @parameter1/base-cms-object-pat getAs Array"

This reverts commit d7385d412fbbdc7e58d27aae3d2f09baa1bbd0b1.
